thrift/src/main/java/com/linecorp/armeria/common/thrift/text/TTextProtocol.java

```diff
@@ -337,7 +344,25 @@ public void writeI16(short i16) throws TException {
 
     @Override
     public void writeI32(int i32) throws TException {
-        writeNameOrValue(TypedParser.INTEGER, i32);
+        if (!writeEnumsAsString) {
```

Comment:
This is to avoid an unnecessary call to `getCurrentFieldClassIfIs()`. I can remove it if you consider the impact insignificant

Reply:
Sounds good to me.